### PR TITLE
Update systemd-oomd man page.

### PR DESCRIPTION
**Pull request by deleriux**
The systemd-oomd documentation currently indicates that eligible candidates need to have both memory.oom.group set to 1 and be a leaf in the cgroup tree.

But according to the code documention and its logic in systemd-oomd: 
```
/* Fill 'new_h' with 'path's descendant OomdCGroupContexts. Only include descendant cgroups that are possible
 * candidates for action. That is, only leaf cgroups or cgroups with memory.oom.group set to "1". */

```
The current wording is misleading. Fix it to say 'or' which is actually the program does.

### CODE REVIEW

**Comment by keszybz** [comment]
Hmmm, "and" is the usual and unambiguous word that is used in such a sentence.
See e.g. https://medium.com/@Ediket/what-s-the-best-way-to-write-a-list-7fb9e0adcef9.
> “And” is inclusive and means all of the items mentioned apply, whereas “or” indicates that only one of the items is relevant.

**Comment by deleriux** [comment]
> "And" is inclusive and and means all of the items mentioned apply

Exacty. Thats wrong. All the items do not apply.

If you read it in the way its written then in order to get a candidate for `systemd-oomd` to target a unit, I would have had to set `OOMPolicy` to `kill` in the unit that should be a candiate _and_ set the `ManagedOOMPreference` to `kill` also in parental slice or scope. 

This isn't how it works or is designed to work.

The logic in the code at `recursively_get_cgroup_context`  in file `src/oom/oomd-manager.c` is that  **one** of the following conditions must be true in order to be a candidate for killing.
- The cgroup is a leaf (has no descendants).
- The cgroup has directly set `memory.oom.group` to `1`.

For the avoidance of doubt. The code says only _one_ item has to be relevant.

So, basically `systemd-oomd` can (when `memory.oom.group` is set to `1`) kill off an entire scope or slice (along with all the units inside them) if `OOMPolicy` is set to `kill` in a scope or a slice. If not however, which is typically the default (at least on my system) `systemd-oomd` can only choose leaf nodes (individual units) to kill.

**Comment by keszybz** [comment]
No. You're mixing up two things: a list of characteristics (adjectives), where indeed we would use "or" because either or both adjectives may apply, with a list of objects (nouns), which is what we have in this case. "All the items do apply" in the sense the all the items in the list are included in the enumeration simultaneously. Please read the example given in the linked blog note; it uses "and" in a very similar case.

Basically: "Property *a* or property *b* must be set, thus set *A* (everything with prop *a*) and set *B* (everything with prop *b*) are included.".  We would use "set *A* or set *B*" if the two sets were somehow exclusive, but they are not, oomd will try to kill members of both sets.

**Comment by deleriux** [comment]
OK. I understand and get it. As a native English speaker with (obviously since I brought this up) a median handle on the English language this still threw me off. I'm sure this subtlety must be worse if you are a non-native English speaker.

I'd argue that man pages need to be written in a manner that leaves little room for interpretation, especially on a casual reading. Given that I had to check the source code to confirm the behaviour probably indicates that it is not well written.

You could use instead of "or" something like "along with" instead?

**Comment by YHNdnzj** [comment]
Hmm, while being a non-native speaker, I don't see the ambiguity in this sentence 🤔

**Comment by deleriux** [comment]
I'm fine to say its probably just me if people disagree! :+1:

**Comment by keszybz** [comment]
OK, let's close this then.
